ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -19,6 +19,8 @@
 using namespace tt::constants;
 using ttnn::operations::unary::UnaryWithParam;
 
+const bool tt::operations::primary::Matmul::disable_stagger_from_env = std::getenv("DISABLE_MATMUL_STAGGER") != nullptr;
```

Comment:
Why do you have an env flag here for this feature when it seems like the param is being passed through from higher level op calls? What happens if op sets `disable_stagger` to true, and user sets `DISABLE_MATMUL_STAGGER=0`? Wouldn't that enable it?

Reply:
We decided to have an option where you could turn off stagger in general, not just per-op, and since it is an opt-out mechanism, we would disable stagger in case any of the two (env var or program config) is set to True. 

I did not add some additional checks in case env var is set to zero for example, we are just checking that it is set, so I will update that.